contracts/BatchReveal.sol

```diff
@@ -88,24 +88,126 @@ abstract contract BatchReveal is
     /// @param _collectionSize Needs to be sent by child contract
+    /// @param _revealStartTime Batch reveal start time
+    /// @param _revealInterval Batch reveal interval
     function initializeBatchReveal(
         uint256 _revealBatchSize,
-        uint256 _collectionSize
+        uint256 _collectionSize,
+        uint256 _revealStartTime,
+        uint256 _revealInterval
     ) internal onlyInitializing {
+        if (_collectionSize == 0) {
+            revert Launchpeg__LargerCollectionSizeNeeded();
+        }
         if (
-            _collectionSize % _revealBatchSize != 0 ||
-            _revealBatchSize == 0 ||
+            (_revealBatchSize != 0 && (_collectionSize % _revealBatchSize != 0)) ||
             _revealBatchSize > _collectionSize
         ) {
             revert Launchpeg__InvalidBatchRevealSize();
         }
+        // We assume that if the reveal is more than 100 days in the future, that's a mistake
+        // Same if the reveal interval is longer than 10 days
+        if (
+            _revealStartTime > block.timestamp + 8_640_000 ||
+            _revealInterval > 864_000
+        ) {
+            revert Launchpeg__InvalidRevealDates();
+        }
         revealBatchSize = _revealBatchSize;
         collectionSize = _collectionSize;
-        _rangeLength = (_collectionSize / _revealBatchSize) * 2;
         intCollectionSize = int128(int256(_collectionSize));
+        if (_revealBatchSize != 0) {
+            _rangeLength = (_collectionSize / _revealBatchSize) * 2;
+        }
+        revealStartTime = _revealStartTime;
+        revealInterval = _revealInterval;
+    }
+
+    /// @notice Set the reveal batch size. Can only be set after
+    /// batch reveal has been initialized and before a batch has
+    /// been revealed.
+    /// @dev Set to 0 to disable batch reveal
+    /// @param _revealBatchSize New reveal batch size
+    function _setRevealBatchSize(uint256 _revealBatchSize)
+        internal
+        initialized
+        revealNotStarted
+    {
+        if (_revealBatchSize == 0) {
+            _rangeLength = 0;
+        } else {
+            if (
+                collectionSize % _revealBatchSize != 0 ||
+                _revealBatchSize > collectionSize
+            ) {
+                revert Launchpeg__InvalidBatchRevealSize();
+            }
+            _rangeLength = (collectionSize / _revealBatchSize) * 2;
+        }
+        revealBatchSize = _revealBatchSize;
+        emit RevealBatchSizeSet(_revealBatchSize);
+    }
+
+    /// @notice Set the batch reveal start time. Can only be set after
+    /// batch reveal has been initialized and before a batch has
+    /// been revealed.
+    /// @param _revealStartTime New batch reveal start time
+    function _setRevealStartTime(uint256 _revealStartTime)
+        internal
+        initialized
+        revealNotStarted
+    {
+        if (_revealStartTime > block.timestamp + 8_640_000) {
```

Comment:
should we also check if the reveal start time is after the current timestamp?

Reply:
Why not? Maybe there is a use case where we want to set it before but I don't see it